cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -15,22 +12,44 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   import configuration.{includeStack, includeThread}
 
+  def verify(recordingFile: RecordingFile): Boolean = {
+
+    val allEventsTypes = recordingFile.readEventTypes();
+
+    val eventTypes = allEventsTypes.stream().map(eventType => eventType.getLabel).collect(Collectors.toSet[String])
+
+    if(!eventTypes.contains("Allocation in new TLAB")){
+      throw new JfrEventNotFoundException("'Allocation in new TLAB' Event not found")
+    }
+    if(!eventTypes.contains("Allocation outside TLAB")){
+      throw new JfrEventNotFoundException("'Allocation outside TLAB' Event not found")
+    }
+    if(!eventTypes.contains("Method Profiling Sample")){
+      throw new JfrEventNotFoundException("'Method Profiling Sample' Event not found")
+    }
+    return true;
+  }
+
   def parse(): Unit = {
 
+
+
     println(s"*** File $file")
     try {
-      val recording = FlightRecordingLoader.loadFile(file)
-      for (t <- recording.getEventTypes.asScala) {
-        println(t.getName)
-      }
-      val view = recording.createView
-      view.asScala.find(_.getEventType.getName == "GC TLAB Configuration") match {
-        case Some(e) => handleTlabConfiguration(e)
-        case _ => ???
+      val recording = new RecordingFile(file.toPath);
+
+      try{
+        verify(recording)
+      }catch{
+        case e: JfrEventNotFoundException => {
```

Comment:
generally there isnt much point in catching the exception to print an error message without the stack trace

Also prefer to use the string interpolator form e.g. s"failed [file]${file.getName}....." in this case 
but if you added the file info to the exception you would not need to catch it and print in the first place :-)

This app is to be used by developers, so there isnt a need to catch the exception, just make sure that the exceptions hav all f the info that they need to help you diagnose the issue